### PR TITLE
Create call for evidence

### PR DESCRIPTION
**Pull request by beccapearce**
https://trello.com/c/Bx2zLsHM/1312-create-the-basic-call-for-evidence-document-type-in-whitehall

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

File: app/models/call_for_evidence_response.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
+class CallForEvidenceResponse < ApplicationRecord
```
Comment (ryanb-gds) [inline app/models/call_for_evidence_response.rb:1]: I think this model should not be part of this PR - to be added in the outcome PR

**Review by ryanb-gds — CHANGES_REQUESTED** [review]
Looking pretty good to me. Suggest removing the `CallForEvidenceResponse` model from the PR and adding in the "Outcome" PR. Also might be nice if the rename of `Response` to `ConsultationResponse` was its own PR, would reduced the difficulty of reviewing this PR a little.

And obvs still needs the feature flag to be applied!

File: app/controllers/admin/editions_controller.rb [hunk 2]
```diff
@@ -269,6 +269,21 @@ def permitted_edition_attributes
...
+          {
+            call_for_evidence_response_form_attributes: [
+              :id,
+              :title,
+              :_destroy,
+              :attachment_action,
+              { call_for_evidence_response_form_data_attributes: %i[id file file_cache] },
+            ],
+          },
+        ],
```
[the remark this answers is not in the document]
Comment (minhngocd) [reply app/controllers/admin/editions_controller.rb:286]: I've added a trello card to track this in the backlog

File: db/migrate/20230617100117_add_call_for_evidence_response_schemas.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    create_table "call_for_evidence_participations", id: :integer, charset: "utf8mb3", force: :cascade do |t|
+      t.integer "edition_id", foreign_key: true
+      t.string "link_url"
+      t.datetime "created_at", precision: nil
+      t.datetime "updated_at", precision: nil
+      t.string "email"
+      t.integer "call_for_evidence_response_form_id"
+      t.text "postal_address"
+      t.index %w[call_for_evidence_response_form_id], name: "index_cons_participations_on_cons_response_form_id"
```
Comment (ryanb-gds) [inline db/migrate/20230617100117_add_call_for_evidence_response_schemas.rb:24]: I don't think this index name is right? "cons" should maybe be "cfes"? What's a bit odd is I think if there's an index with the same name on the consultation participation table the migration should fail, but tests suggest that isn't happening 🤔

File: db/schema.rb [hunk 4]
```diff
@@ -59,6 +59,31 @@
...
 
+  create_table "call_for_evidence_participations", id: :integer, charset: "utf8mb3", force: :cascade do |t|
+    t.integer "edition_id"
+    t.string "link_url"
+    t.datetime "created_at", precision: nil
+    t.datetime "updated_at", precision: nil
+    t.string "email"
+    t.integer "call_for_evidence_response_form_id"
+    t.text "postal_address"
+    t.index ["call_for_evidence_response_form_id"], name: "index_cons_participations_on_cons_response_form_id"
```
Comment (ryanb-gds) [inline db/schema.rb:70]: Same problem as above - "cons" in index name

**Review by ryanb-gds — COMMENTED** [review]
Essentially ready to merge but I think it's worth checking that potential duplicate index name?

File: db/schema.rb (hunk 4 above)
Comment (minhngocd) [reply db/schema.rb]: fixed

**Review by ryanb-gds — APPROVED** [review]
